kong/core/globalpatches.lua

```diff
@@ -272,11 +272,11 @@ return function(options)
       -- Because the CLI runs in `xpcall`, we cannot use yielding cosockets.
       -- Hence, we need to stick to luasocket when using cassandra or pgmoon
       -- in the CLI.
-      for _, namespace in ipairs({"cassandra", "pgmoon-mashape"}) do
-        local socket = require(namespace .. ".socket")
-        socket.force_luasocket(ngx.get_phase(), true)
-      end
+      local cassandra_socket = require "cassandra.socket"
+      cassandra_socket.force_luasocket(ngx.get_phase(), true)
 
+      local psql_dao = require "kong.dao.db.postgres"
+      psql_dao.force_luasocket(ngx.get_phase(), true)
```

Comment:
Because we got rid of the migrations running in the non-yieldable `require`, and because we use LuaJIT which supports resumable `xpcall`, we should actually be able to get rid of this hack altogether, and reduce our `globalpatches` size (the following patch removes the first branch for `options.cli` and reindents the cosocket patch:

```diff
diff --git a/kong/core/globalpatches.lua b/kong/core/globalpatches.lua
index e366bca3..cbf38388 100644
--- a/kong/core/globalpatches.lua
+++ b/kong/core/globalpatches.lua
@@ -267,93 +267,81 @@ return function(options)
 
 
   do -- cosockets connect patch for dns resolution for: cli, rbusted and OpenResty
+    local sub = string.sub
 
-    if options.cli then
-      -- Because the CLI runs in `xpcall`, we cannot use yielding cosockets.
-      -- Hence, we need to stick to luasocket when using cassandra or pgmoon
-      -- in the CLI.
-      local cassandra_socket = require "cassandra.socket"
-      cassandra_socket.force_luasocket(ngx.get_phase(), true)
-
-      local psql_dao = require "kong.dao.db.postgres"
-      psql_dao.force_luasocket(ngx.get_phase(), true)
-    else
-      local sub = string.sub
-
-      --- Patch the TCP connect and UDP setpeername methods such that all
-      -- connections will be resolved first by the internal DNS resolver.
-      -- STEP 1: load code that should not be using the patched versions
-      require "resty.dns.resolver" -- will cache TCP and UDP functions
-
-      -- STEP 2: forward declaration of locals to hold stuff loaded AFTER patching
-      local toip
-
-      -- STEP 3: store original unpatched versions
-      local old_tcp = ngx.socket.tcp
-      local old_udp = ngx.socket.udp
-
-      local old_tcp_connect
-      local old_udp_setpeername
-
-      local function tcp_resolve_connect(sock, host, port, sock_opts)
-        local target_ip, target_port = toip(host, port)
-        if not target_ip then
-          return nil, "[toip() name lookup failed]: " .. tostring(target_port) -- err
-        end
+    --- Patch the TCP connect and UDP setpeername methods such that all
+    -- connections will be resolved first by the internal DNS resolver.
+    -- STEP 1: load code that should not be using the patched versions
+    require "resty.dns.resolver" -- will cache TCP and UDP functions
 
-        -- need to do the extra check here: https://github.com/openresty/lua-nginx-module/issues/860
-        if not sock_opts then
-          return old_tcp_connect(sock, target_ip, target_port)
-        end
+    -- STEP 2: forward declaration of locals to hold stuff loaded AFTER patching
+    local toip
 
-        return old_tcp_connect(sock, target_ip, target_port, sock_opts)
-      end
+    -- STEP 3: store original unpatched versions
+    local old_tcp = ngx.socket.tcp
+    local old_udp = ngx.socket.udp
 
-      local function udp_resolve_setpeername(sock, host, port)
-        local target_ip, target_port
+    local old_tcp_connect
+    local old_udp_setpeername
 
-        if sub(host, 1, 5) == "unix:" then
-          target_ip = host -- unix domain socket, so just maintain the named values
+    local function tcp_resolve_connect(sock, host, port, sock_opts)
+      local target_ip, target_port = toip(host, port)
+      if not target_ip then
+        return nil, "[toip() name lookup failed]: " .. tostring(target_port) -- err
+      end
 
-        else
-          target_ip, target_port = toip(host, port)
+      -- need to do the extra check here: https://github.com/openresty/lua-nginx-module/issues/860
+      if not sock_opts then
+        return old_tcp_connect(sock, target_ip, target_port)
+      end
 
-          if not target_ip then
-            return nil, "[toip() name lookup failed]: " .. tostring(target_port) -- err
-          end
-        end
+      return old_tcp_connect(sock, target_ip, target_port, sock_opts)
+    end
 
-        return old_udp_setpeername(sock, target_ip, target_port)
-      end
+    local function udp_resolve_setpeername(sock, host, port)
+      local target_ip, target_port
+
+      if sub(host, 1, 5) == "unix:" then
+        target_ip = host -- unix domain socket, so just maintain the named values
 
-      -- STEP 4: patch globals
-      _G.ngx.socket.tcp = function(...)
-        local sock = old_tcp(...)
+      else
+        target_ip, target_port = toip(host, port)
 
-        if not old_tcp_connect then
-          old_tcp_connect = sock.connect
+        if not target_ip then
+          return nil, "[toip() name lookup failed]: " .. tostring(target_port) -- err
         end
+      end
 
-        sock.connect = tcp_resolve_connect
+      return old_udp_setpeername(sock, target_ip, target_port)
+    end
+
+    -- STEP 4: patch globals
+    _G.ngx.socket.tcp = function(...)
+      local sock = old_tcp(...)
 
-        return sock
+      if not old_tcp_connect then
+        old_tcp_connect = sock.connect
       end
 
-      _G.ngx.socket.udp = function(...)
-        local sock = old_udp(...)
+      sock.connect = tcp_resolve_connect
 
-        if not old_udp_setpeername then
-          old_udp_setpeername = sock.setpeername
-        end
+      return sock
+    end
 
-        sock.setpeername = udp_resolve_setpeername
+    _G.ngx.socket.udp = function(...)
+      local sock = old_udp(...)
 
-        return sock
+      if not old_udp_setpeername then
+        old_udp_setpeername = sock.setpeername
       end
 
-      -- STEP 5: load code that should be using the patched versions, if any (because of dependency chain)
-      toip = require("resty.dns.client").toip  -- this will load utils and penlight modules for example
+      sock.setpeername = udp_resolve_setpeername
+
+      return sock
     end
+
+    -- STEP 5: load code that should be using the patched versions, if any (because of dependency chain)
+    toip = require("resty.dns.client").toip  -- this will load utils and penlight modules for example
   end
 end
```